### ISSUE TITLE
sudden static compilation issue on jrei/crystal-alpine... is there a better docker image I should use?

### ISSUE THREAD

**Issue by sam0x17** [crystal-lang/crystal#6276]
This started happening yesterday AFAIK. Before yesterday, doing the following resulted in a nice static linux binary on any linux system so long as docker is set up to run as non-root and the daemon is running:
```
docker pull jrei/crystal-alpine && docker run --rm -it -v $PWD:/app -w /app jrei/crystal-alpine crystal build src/*.cr -o my_app --release --static --no-debug
```

As of yesterday this no longer works and I am getting:
```
/usr/lib/libcrypto.a(c_zlib.o): In function `zlib_stateful_expand_block':
/home/buildozer/aports/main/openssl/src/openssl-1.0.2o/crypto/comp/c_zlib.c:239: undefined reference to `inflate'
/usr/lib/libcrypto.a(c_zlib.o): In function `zlib_stateful_compress_block':
/home/buildozer/aports/main/openssl/src/openssl-1.0.2o/crypto/comp/c_zlib.c:210: undefined reference to `deflate'
/usr/lib/libcrypto.a(c_zlib.o): In function `bio_zlib_free':
/home/buildozer/aports/main/openssl/src/openssl-1.0.2o/crypto/comp/c_zlib.c:504: undefined reference to `inflateEnd'
/home/buildozer/aports/main/openssl/src/openssl-1.0.2o/crypto/comp/c_zlib.c:509: undefined reference to `deflateEnd'
/usr/lib/libcrypto.a(c_zlib.o): In function `zlib_stateful_finish':
/home/buildozer/aports/main/openssl/src/openssl-1.0.2o/crypto/comp/c_zlib.c:187: undefined reference to `inflateEnd'
/home/buildozer/aports/main/openssl/src/openssl-1.0.2o/crypto/comp/c_zlib.c:188: undefined reference to `deflateEnd'
/usr/lib/libcrypto.a(c_zlib.o): In function `zlib_stateful_init':
/home/buildozer/aports/main/openssl/src/openssl-1.0.2o/crypto/comp/c_zlib.c:157: undefined reference to `inflateInit_'
/home/buildozer/aports/main/openssl/src/openssl-1.0.2o/crypto/comp/c_zlib.c:168: undefined reference to `deflateInit_'
/usr/lib/libcrypto.a(c_zlib.o): In function `bio_zlib_flush':
/home/buildozer/aports/main/openssl/src/openssl-1.0.2o/crypto/comp/c_zlib.c:675: undefined reference to `deflate'
/home/buildozer/aports/main/openssl/src/openssl-1.0.2o/crypto/comp/c_zlib.c:680: undefined reference to `zError'
/usr/lib/libcrypto.a(c_zlib.o): In function `bio_zlib_write':
/home/buildozer/aports/main/openssl/src/openssl-1.0.2o/crypto/comp/c_zlib.c:596: undefined reference to `deflateInit_'
/home/buildozer/aports/main/openssl/src/openssl-1.0.2o/crypto/comp/c_zlib.c:630: undefined reference to `deflate'
/home/buildozer/aports/main/openssl/src/openssl-1.0.2o/crypto/comp/c_zlib.c:633: undefined reference to `zError'
/usr/lib/libcrypto.a(c_zlib.o): In function `bio_zlib_read':
/home/buildozer/aports/main/openssl/src/openssl-1.0.2o/crypto/comp/c_zlib.c:535: undefined reference to `inflateInit_'
/home/buildozer/aports/main/openssl/src/openssl-1.0.2o/crypto/comp/c_zlib.c:546: undefined reference to `inflate'
/home/buildozer/aports/main/openssl/src/openssl-1.0.2o/crypto/comp/c_zlib.c:549: undefined reference to `zError'
collect2: error: ld returned 1 exit status
Error: execution of command failed with code: 1: `cc "${@}" -o '/app/gcf'  -rdynamic -static  `command -v pkg-config > /dev/null && pkg-config --libs --silence-errors libssl || printf %s '-lssl -lcrypto'` `command -v pkg-config > /dev/null && pkg-config --libs --silence-errors libcrypto || printf %s ' -lcrypto'` /usr/lib/libpcre.a /usr/lib/libgc.a /usr/lib/libpthread.a /usr/lib/crystal/core/ext/libcrystal.a /usr/lib/libevent.a /usr/lib/librt.a -L/usr/lib -L/usr/local/lib`
```

`zlib-dev` is included in the Dockerfile, so I don't see why this would happen on a musl-based system.

I checked the jrei/crystal-alpine repo (https://hub.docker.com/r/jrei/crystal-alpine/) and no changes have been made to the Dockerfile in recent history. The only two explanations I can think of are that a change was made to alpine:edge that has removed something that zlib needs, or a change was made to the crystal alpine package in APK that changed the zlib requirements.

Is there a better docker image I should be using? I am looking for something I can reliably compile static binaries with based on latest crystal.

**Comment by RX14** [comment]
Looks like a bug in alpine linux, please report it there.

**Comment by sam0x17** [comment]
TLDR: don't use automated daily builds because stuff will break

**Comment by j8r** [comment]
This is expected because the the image is Alpine Edge, a development branch that may break at any time.
When Alpine 3.8 will be out as the latest stable release, we will be able to use it with a lot more confidence. It will include Crystal 0.25 along with other libraries that include the `.a` archives for static linking.
An image that was tested to work can also be tagged locally to be able to reuse it.
